client/src/modules/shopping-list/model/reducer.js

```diff
@@ -16,6 +16,18 @@ const shoppingLists = (state = {}, action) => {
       const { [action.payload]: removed, ...newState } = state;
       return newState;
     }
+    case ShoppingListActionTypes.UPDATE_SUCCESS: {
+      const updateList = {
+        ...state[action.payload.listId]
+      };
+      updateList.name = action.payload.name || updateList.name;
+      updateList.description =
+        action.payload.description || updateList.description;
+      return {
+        ...state,
+        [action.payload.listId]: updateList
+      };
+    }
```

Comment:
maybe this would easier to reason about ??
```
      const previousName = { ...state[action.payload.listId] }.name;
      const previousDescription = { ...state[action.payload.listId] }
        .description;

      const updatedList = {
        ...state[action.payload.listId],
        name: action.payload.name || previousName,
        description: action.payload.description || previousDescription
      };

      return {
        ...state,
        [action.payload.listId]: updatedList
      };
```

Reply:
Good idea to make it more readable. I changed it similarly to your proposition but avoiding multiple spreading